extension/lib/Phases.jsm

```diff
@@ -89,39 +98,46 @@ this.Phases = {
       Pioneer.utils.endStudy();
       return;
     }
-
-    if (phase.surveyURL) {
-      this.showSurvey(phase.surveyURL);
-    }
   },
 
   /**
    * Prompts the user to take a survey.
    */
   showSurvey() {
-    // TODO: show doorhanger instead of opening tab directly
-    // TODO: Only show a survey once per ${INTERVAL}.
-
     const state = State.load();
     const phase = this.getCurrentPhase();
 
     if (!state.promptsRemaining.hasOwnProperty(state.phaseName)) {
       state.promptsRemaining[state.phaseName] = phase.promptRepeat || 3;
     }
 
-    if (state.promptsRemaining[state.phaseName] > 0) {
-      state.promptsRemaining[state.phaseName] -= 1;
-      State.save(state);
+    const hasPromptsRemaining = state.promptsRemaining[state.phaseName] > 0;
+
+    if (!state.lastSurveyShown.hasOwnProperty(state.phaseName)) {
+      state.lastSurveyShown[state.phaseName] = 0;
+    }
+
+    const timesinceSurveyShown = Date.now() - state.lastSurveyShown[state.phaseName];
+    const shouldShowDoorhanger = timesinceSurveyShown > Config.showDoorhangerInterval;
+
+    if (shouldShowDoorhanger && hasPromptsRemaining) {
       const recentWindow = RecentWindow.getMostRecentBrowserWindow({
         private: false,
         allowPopups: false,
       });
+
       if (recentWindow && recentWindow.gBrowser) {
-        // TODO: add pioneerID and utm_source parameters to surveyURL
-        const tab = recentWindow.gBrowser.loadOneTab(phase.surveyURL, { inBackground: false });
+        const pioneerId = Pioneer.utils.getPioneerId();
+        const queryString = `utm_source=pioneer&utm_campaign=online-news&pioneer_id=${pioneerId}`;
+        const surveyURL = `${phase.surveyURL}?${queryString}`;
```

Comment:
This should probably use the `URL` constructor:

```js
const surveyURL = new URL(phase.surveyURL);
surveyURL.searchParams.set("utm_source", "pioneer');
surveyURL.searchParams.set("utm_campaign", "online-news");
surveyURL.searchParams.set("pioneer_id", pioneerId);
```

More verbose, but safer because it correctly handles existing query strings.